bok_choy/page_object.py

```diff
@@ -338,6 +338,20 @@ def wait_for_page(self, timeout=30):
         Raises:
             BrokenPromise: The timeout is exceeded without the page loading successfully.
         """
+        def _is_document_ready():
+            """
+            Check the loading state of the document to ensure the document
+            and all sub-resources have finished loading.
+            """
+            return self.browser.execute_script(
+                "return document.readyState=='complete'")
+
+        EmptyPromise(
```

Comment:
I could go either way, but I guess I used an EmptyPromise here to be similar to the Promise in the next statement below.